java/testing/testing-ii/testing-private-methods-and-fields.md

```diff
@@ -24,48 +18,84 @@ aspects:
 
 Times may come when you will have to test **private** methods. As using *inner classes* to write tests is *not advisable* (as tests would likely end up in production), three other methods are to be considered.
 
-The least intrusive method is to test **the contract** that class has with other objects: is every other object for which the private method was used behaving properly? If they do, then you can assume the code works as expected.
+The least intrusive method is to test **the contract** that class has with other objects: is every other object for which the private method was used behaving properly? If they do, then you can assume the code works as expected. With this approach, you can test the side-effects of your private function.
 
-Another option is to use **reflection**.  Reflection can be used for observing and modifying program execution. It allows for instantiation of objects and invocation of methods that would otherwise be private:
+Another option is to use **reflection**. Reflection can be used for observing and modifying program execution. It allows for instantiation of objects and invocation of methods that would otherwise be private.
+
+Say your class is similar to `User` in that it has a private method that you want to test:
+
+```java
+class User {
+  String name;
+
+  public User(String newName) {
+    name = newName;
+  }
+
+  private boolean checkHashPass(String hash) {
+    String userHashedPass = "...";
+    return userHashedPass.equals(hash);
+  }
+}
+```
+
+You can access the method to be tested like so:
 
 ```java
 import java.lang.reflect.Method;
 
-User john = new User("John");
+// build an argument of type:
```

Comment:
maybe say something like

> to target the private method you need to match its arguments signature
for this we instantiate an array of `Classes`, with its first and only item a `String` class. our method only accepts a string as input